### PR TITLE
Catch syntax errors in custom visualisations

### PR DESCRIPTION
**Pull request by s9ferech**
### Pull Request Description

This is a potential solution to #1310.

Syntax errors in custom visualisations caused panics during IDE startup. Now they are caught and handled like errors during execution of the visualisation code, which means that we write a log message to the console.

### Important Notes

The wasm-bindgen API that we used to parse visualisation code from JS, does not allow to handle syntax errors, it always panics on those. I filed an issue at their GitHub repo, suggesting that they allow error handling. As a temporary fix, this PR includes a copy of their relevant code with the necessary changes. We could merge this now or wait to see if they react to the issue and provide the possibility for a cleaner solution.

With this PR, syntax errors in visualisations will only be logged to the console, which means that they stay invisible to normal users. In the long term, they should be reported on the GUI.

### Checklist
Please include the following checklist in your PR:

- [x] The `CHANGELOG.md` was updated with the changes introduced in this PR.
- [ ] The documentation has been updated if necessary.
- [x] All code conforms to the [Rust](https://github.com/enso-org/enso/blob/main/docs/style-guide/rust.md) style guide.
- [ ] All code has automatic tests where possible.
- [ ] All code has been profiled where possible.
- [x] All code has been manually tested in the IDE.
- [ ] All code has been manually tested in the "debug/interface" scene.
- [ ] All code has been manually tested by the PR owner against our [test scenarios](https://docs.google.com/spreadsheets/d/1RatJDM_f9_3bvYhl3Bpq2d8SyKgtVdrV1RkGxPU17c8/edit?ts=5faa7049#gid=0).
- [ ] All code has been manually tested by at least one reviewer against our [test scenarios](https://docs.google.com/spreadsheets/d/1RatJDM_f9_3bvYhl3Bpq2d8SyKgtVdrV1RkGxPU17c8/edit?ts=5faa7049#gid=0).

### CODE REVIEW

File: src/rust/ide/view/graph-editor/src/component/visualization/foreign/java_script/definition/function.rs [hunk 1]
```diff
@@ -0,0 +1,156 @@
+//! The whole content of this file is copied from `js_sys` just to modify `new_with_args and `new_no_args` such that they catch syntax errors.
+//! This is supposed to be temporary solution.
+//! I opened a GitHub issue suggesting that this will be included in wasm-bindgen (https://github.com/rustwasm/wasm-bindgen/issues/2496).
```
Comment (wdanilo) [inline src/rust/ide/view/graph-editor/src/component/visualization/foreign/java_script/definition/function.rs]: Please keep all ouf our lines at max 100 chars long.

File: src/rust/ide/view/graph-editor/src/component/visualization/foreign/java_script/definition/function.rs [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    /// Returns the `Function` value of this JS value if it's an instance of a
+    /// function.
+    ///
+    /// If this JS value is not an instance of a function then this returns
+    /// `None`.
+    #[deprecated(note = "recommended to use dyn_ref instead which is now equivalent")]
+    pub fn try_from(val: &JsValue) -> Option<&Function> {
+        val.dyn_ref()
+    }
+}
```
Comment (wdanilo) [inline src/rust/ide/view/graph-editor/src/component/visualization/foreign/java_script/definition/function.rs]: missing nl

**Review by wdanilo — APPROVED** [review]
I love this. We can merge it now. Would you be so nice and open also PR in the wasm-bindgen repo and link the PR in this `function.rs` doc comment that you've created, please?

Please ask someone from the team to test it as well before the merge, I cant handle it right now.

**Comment by wdanilo** [comment]
Also, before merging this, please test this on the issue that Maciek had: https://github.com/enso-org/ide/issues/1310 

The issue was NOT about panic - it was about hanging gui on startup, while still you were able to drag nodes etc. Just test it fixes this thing. @BinarySoftware please test it too.

**Comment by s9ferech** [comment]
I created the PR at wasm-bindgen. And yes, this solves the original issue.

**Pull request merged**